cpp/dolfinx/fem/CoordinateElement.cpp

```diff
@@ -49,12 +47,20 @@ void CoordinateElement::push_forward(
     const Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic, Eigen::Dynamic,
                                         Eigen::RowMajor>>& cell_geometry) const
 {
-  assert(_compute_physical_coordinates);
+  assert(_finite_element);
   assert(x.rows() == X.rows());
   assert(x.cols() == _gdim);
   assert(X.cols() == _tdim);
-  _compute_physical_coordinates(x.data(), X.rows(), X.data(),
-                                cell_geometry.data());
+
+  // Compute physical coordinates
+  Eigen::Tensor<double, 3, Eigen::RowMajor> phi(1, X.rows(),
+                                                cell_geometry.rows());
+  Eigen::Map<
+      Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>
+      phi_mat(phi.data(), X.rows(), cell_geometry.rows());
+
+  _finite_element->evaluate_reference_basis(phi, X);
```

Comment:
Is it more natural to do this operation before declaring phi_mat, or am I missing something fundamental?

Reply:
I like to keep the declaration and map next to each other. To be honest the `Eigen::Tensor` is a bit of a pain when extracting sub-matrices.